interface/ceed-elemrestriction.c

```diff
@@ -630,6 +630,52 @@ int CeedElemRestrictionApply(CeedElemRestriction rstr, CeedTransposeMode t_mode,
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Restrict an L-vector to an E-vector or apply its transpose ignoring any
+         provided orientations
+
+  @param[in]  rstr    CeedElemRestriction
+  @param[in]  t_mode  Apply restriction or transpose
+  @param[in]  u       Input vector (of size @a l_size when t_mode=@ref CEED_NOTRANSPOSE)
+  @param[out] ru      Output vector (of shape [@a num_elem * @a elem_size] when t_mode=@ref CEED_NOTRANSPOSE).
+                        Ordering of the e-vector is decided by the backend.
+  @param[in]  request Request or @ref CEED_REQUEST_IMMEDIATE
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref User
+**/
+int CeedElemRestrictionApplyUnsigned(CeedElemRestriction rstr, CeedTransposeMode t_mode, CeedVector u, CeedVector ru, CeedRequest *request) {
+  CeedInt m, n;
+
+  if (t_mode == CEED_NOTRANSPOSE) {
+    m = rstr->num_blk * rstr->blk_size * rstr->elem_size * rstr->num_comp;
+    n = rstr->l_size;
+  } else {
+    m = rstr->l_size;
+    n = rstr->num_blk * rstr->blk_size * rstr->elem_size * rstr->num_comp;
+  }
+  if (n != u->length) {
+    // LCOV_EXCL_START
+    return CeedError(rstr->ceed, CEED_ERROR_DIMENSION,
+                     "Input vector size %" CeedInt_FMT " not compatible with element restriction (%" CeedInt_FMT ", %" CeedInt_FMT ")", u->length, m,
+                     n);
+    // LCOV_EXCL_STOP
+  }
+  if (m != ru->length) {
+    // LCOV_EXCL_START
+    return CeedError(rstr->ceed, CEED_ERROR_DIMENSION,
+                     "Output vector size %" CeedInt_FMT " not compatible with element restriction (%" CeedInt_FMT ", %" CeedInt_FMT ")", ru->length,
+                     m, n);
+    // LCOV_EXCL_STOP
+  }
```

Comment:
New approach with `CeedCheck` macro:
```suggestion
   CeedCheck(n == u->length, rstr->ceed, CEED_ERROR_DIMENSION,
                     "Input vector size %" CeedInt_FMT " not compatible with element restriction (%" CeedInt_FMT ", %" CeedInt_FMT ")", u->length, m,
                     n);
  CeedCheck(m == ru->length, rstr->ceed, CEED_ERROR_DIMENSION,
                     "Output vector size %" CeedInt_FMT " not compatible with element restriction (%" CeedInt_FMT ", %" CeedInt_FMT ")", ru->length,
                     m, n);
```

Reply:
Missed that one during rebase, fixed in [57d0363](https://github.com/CEED/libCEED/pull/1196/commits/57d03632a4275bf84e2c35de7a183a4fc8e95d18).